azurerm/internal/services/containers/kubernetes_cluster_pod_identity_resource.go

```diff
@@ -0,0 +1,355 @@
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"identity_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: msiValidate.UserAssignedIdentityID,
+						},
+					},
+				},
+			},
+
+			"exception": {
+				Type:         schema.TypeSet,
+				Optional:     true,
+				AtLeastOneOf: []string{"pod_identity", "exception"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"namespace": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"pod_labels": {
+							Type:     schema.TypeMap,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesClusterPodIdentityCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.KubernetesClustersClient
+	userAssignedIdentitiesClient := meta.(*clients.Client).MSI.UserAssignedIdentitiesClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Managed Kubernetes Cluster Pod Identity.")
+
+	clusterId, err := parse.ClusterID(d.Get("cluster_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id := parse.NewClusterID(subscriptionId, clusterId.ResourceGroup, clusterId.ManagedClusterName)
+	existing, err := client.Get(ctx, id.ResourceGroup, id.ManagedClusterName)
+	if err != nil {
+		return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+	}
+
+	if existing.ManagedClusterProperties == nil {
+		return fmt.Errorf("`ManagedClusterProperties` is nil for %s: %s", id, err)
+	}
+
+	if d.IsNewResource() {
+		if existing.ManagedClusterProperties.PodIdentityProfile != nil &&
+			existing.ManagedClusterProperties.PodIdentityProfile.Enabled != nil &&
+			*existing.ManagedClusterProperties.PodIdentityProfile.Enabled {
+			return tf.ImportAsExistsError("azurerm_kubernetes_cluster_pod_identity", id.ID())
+		}
+	}
+
+	podIdentities, err := expandKubernetesPodIdentities(ctx, userAssignedIdentitiesClient, d.Get("pod_identity").(*schema.Set).List())
+	if err != nil {
+		return err
+	}
+	existing.ManagedClusterProperties.PodIdentityProfile = &containerservice.ManagedClusterPodIdentityProfile{
+		Enabled:                        utils.Bool(true),
+		UserAssignedIdentities:         podIdentities,
+		UserAssignedIdentityExceptions: expandKubernetesPodIdentityExceptions(d.Get("exception").(*schema.Set).List()),
+	}
+
+	if existing.NetworkProfile != nil && existing.NetworkProfile.NetworkPlugin == containerservice.Kubenet {
```

Comment:
`network_plugin` is a `forcenew` property. I have asked the service team why need this property `AllowNetworkPluginKubenet`, they says they want to make users double confirm when using `Kubenet`, because there is some matters that needs to take consideration.

Reply:
Does that mean the `AllowNetworkPluginKubenet` can be `false` even if the `NetworkPlugin` is set to kubenet? If so, then we shall better follow what service team is suggesting to export this property to users, rather than making the decision by them?

Reply:
if `NetworkPlugin` is `kubenet` and users want to support pod identity, `AllowNetworkPluginKubenet` must be `true`. It's just serve for double confirm.